linker/shared/src/test/scala/org/scalajs/linker/IncrementalTest.scala

```diff
@@ -107,9 +225,12 @@ object IncrementalTest {
       classDefs1 = minilib ++ classDefs(backward).map(MemClassDefIRFile(_))
       inits0 = moduleInitializers(backward)
       inits1 = moduleInitializers(!backward)
-      _ <- linkerInc.link(classDefs0, inits0, outputInc, logger)
-      reportInc <- linkerInc.link(classDefs1, inits1, outputInc, logger)
       reportBatch <- linkerBatch.link(classDefs1, inits1, outputBatch, logger)
+      firstReportInc <- linkerInc.link(classDefs0, inits0, outputInc, logger)
+      _ = assertFalse("Sanity check: outputs are supposed to become different",
```

Comment:
I'm not convinced this is worth the additional duplication it brings for the equality comparisons. If you feel strongly, we might want to start using (hamcrest) matchers. (We seem to have some of them implemented for JS).

Reply:
I reverted this.